lib/new_relic/agent/instrumentation/active_record_subscriber.rb

```diff
@@ -91,21 +91,26 @@ def active_record_config(payload)
           end
 
           return unless connection_id = payload[:connection_id]
-          connection = nil
 
           ::ActiveRecord::Base.connection_handler.connection_pool_list.each do |handler|
-            # RE: https://github.com/newrelic/newrelic-ruby-agent/issues/507
-            spec_config = handler.respond_to?(:spec) && handler.spec && handler.spec.config
-            return spec_config if using_makara?(spec_config)
-
-            connection = handler.connections.detect do |conn|
-              conn.object_id == connection_id
-            end
-
-            break if connection
+            connection = handler.connections.detect { |conn| conn.object_id == connection_id }
+            return connection.instance_variable_get(:@config) if connection
+
+            # when using makara, handler.connections will be empty, so use the
+            # spec config instead.
+            # https://github.com/newrelic/newrelic-ruby-agent/issues/507
+            # thank you @lucasklaassen
+            return handler.spec.config if use_spec_config?(handler)
           end
 
-          connection.instance_variable_get(:@config) if connection
+          nil
+        end
+
+        def use_spec_config?(handler)
```

Comment:
This is pretty much `using_makara?` renamed with an additional `respond_to?` check. It's possible that the `#respond_to?` is only needed for the tests and not in real world scenarios, but I'm choosing to err on the safe side by including the check for now.

Reply:
I'm a little worried this method's relation to Makara is buried by naming it `use_spec_config?` What are your thoughts about renaming it to `use_makara_spec_config?`? As it's written right now, the `makara` string in the final clause leads me to believe it can only be used in that context.

Reply:
I very much wanted the method to be generic and not specific to Makara. If we fail to obtain a match from `handler.connections` (which will be an empty array in the case of Makara), but we happen to be able to call `.spec.config` on the `handler` object instead, then that should be all we need to create a segment from. I could foresee other gems also working with this `.spec.config` approach, and so I went with the generic `use_spec_config?` name.

I am envisioning in the future us being able to either drop the Makara specific check from `use_spec_config?` or having it "or"ed with another gem specific check. I decided to leave the Makara specific check in place just because it's currently the only gem we support going to the spec config for.

Both this `use_spec_config?` method and the `bare_adapter_name` method currently contain Makara specific logic, with the intention being that that logic can be added to over time. This gives us easily accessible chokepoints for all gems.